### DISCUSSION TITLE
Como descobrir o valor real que um vendedor pagou por um envio?

### DISCUSSION THREAD

**Discussion by rodurma** [rodurma/forum-meli#1] [Envios (Shipments)]
Olá pessoal, tudo bem?

Temos o endpoint para pegar detalhes de um envio (shipment) como no exemplo abaixo:

```
GET: https://api.mercadolibre.com/shipments/999999999
````

E no `JSON` de resposta temos algumas chaves com valores monetários como por exemplo:

- `base_cost`
- `shipping_option.list_cost`

Sabemos que podem existir as situações onde:

- Comprador paga todo o frete
- Comprador paga uma parte do frete e o vendedor paga o restante
- Comprador não paga frete e o vendedor banca este frete
- Comprador não paga frete e o vendedor banca uma parte e o Mercado Livre outra parte

Como saber de fato o que o vendedor teve de desconto no valor total do pedido relacionado a envio?

Obrigado.

**Comment by tiujhou** [comment]
Na api são 6 tipos de fretes em um anúncio... Custom, not especified, me1 e me2... 

Custom é fixo o frete...
Not: ele pode não ser especificado ou frete grátis
Me1: envio por conta do cliente ou envio grátis.
Me2: nunca tive acesso

Tenta buscar ou no anúncio ou na venda qual dos 4 foi usado, e depois faz o cálculo... Somente ML1 com frete gratis cobra... Tem um gancho que você puxa este dado, o quanto é cobrado em determinado anúncio... Mas isso deve constar na venda...

**Reply by rodurma** [reply]
Então o valor deve vir dessa api que comentei o que queria saber era qual usar.

**Comment by radioative** [comment]
opa, alguem conseguiu descobrir a solução?

**Comment by fialkalp** [comment]
Se a premissa não for partir do endpoint shipment, você consegue pelo endpoint da order. La vc tem o valor liquido de um determinado pedido e, consequentemente, os descontos.


https://api.mercadolibre.com/orders/$ORDER_ID/ 
aqui vc consegue o total_amount e o sale_fee (taxa do produto)

https://api.mercadolibre.com/orders/$ORDER_ID/shipments você encontra todos os dados de envios de uma order
no objeto shipments_options tem os parametros cost (que é o custo de frete pago pelo comprador) e o list_cost (valor pago pelo vendedor)

**Comment by fialkalp** [comment]
@rodurma vc sabe a definição do base_cost?
